src/components/OSCALMetadata.js

```diff
@@ -33,8 +37,131 @@ const useStyles = makeStyles((theme) => ({
 // Returns a string with a locality-sensitive representation of this date
 const formatDate = (isoDate) => new Date(isoDate).toLocaleString();
 
+function saveModifiedMetadata(modifiedField, modifiableMetadata, newValue) {
+  /* Since the browser url will be of the form
+   * http://localhost:8080/system-security-plans?url=http://localhost:8080/oscal/v1/ssps/{uuid}
+   * and we just want the URL defined in the url parameter of the browser url, we must find the
+   * location of the ? and increment that value by 5 to get the start of the desired url.
+   */
+  const startOfUrl = window.location.href.indexOf("?") + 5;
+  const url = window.location.href.substring(startOfUrl);
+
+  /* We must specify a method of PATCH so the backend service knows we are just
+   * updating a value.
+   *
+   * We also provide a body so we can specify what needs to be modified.
+   */
+  fetch(url, {
+    method: "PATCH",
+    body: JSON.stringify({
+      metadata: {
+        modifiedField: newValue,
+      },
+    }),
+  })
+    .then((res) => res.json())
+    .then(
+      (result) => {
+        /* If we reach this point, then the OSCAL file has successfully been modified
+         * and we want those changes to be visible to the user.
+         */
+        modifiableMetadata["last-modified"][1](
+          formatDate(result["last-modified"])
+        );
+        modifiableMetadata[modifiedField].value[1](newValue);
+      },
+      () => {
+        alert(
+          `Could not update the ${modifiedField} metadata field with value: ${newValue}.`
+        );
+      }
+    );
+}
+
+function displayEditableTextField(modifiedField, modifiableMetadata) {
+  return modifiableMetadata[modifiedField].edit[0] ? (
+    <Typography variant={modifiedField === "title" ? "h6" : "body2"}>
+      <TextField
+        size="medium"
+        variant="outlined"
+        inputRef={modifiableMetadata[modifiedField].ref}
+      >
+        {modifiableMetadata[modifiedField].value[0]}
+      </TextField>
+    </Typography>
+  ) : (
+    <Typography variant={modifiedField === "title" ? "h6" : "body2"}>
+      {modifiableMetadata[modifiedField].value[0]}
+    </Typography>
+  );
+}
+
+function displayEditIcons(modifiedField, modifiableMetadata) {
```

Comment:
This feels like it should be its own lower-level component that is only responsible for telling some other component to enter into editing mode.